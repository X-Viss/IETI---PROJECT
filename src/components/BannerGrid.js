import React from 'react';
import GridListTileBar from "@material-ui/core/GridListTileBar";
import IconButton from "@material-ui/core/IconButton";
import InfoIcon from "@material-ui/icons/Info";

function BannerGrid(props) {
    return (
        <div>
            <img src={props.tile.img} alt={props.tile.title}/>
            {props.tile.title && props.tile.author ?
                <GridListTileBar
                    title={props.tile.title}
                    subtitle={<span>by: {props.tile.author}</span>}
                    actionIcon={
                        <IconButton aria-label={`info about ${props.tile.title}`} className={props.classes.icon}
                                    onClick={event => {
                                        event.preventDefault();
                                        window.location.href = "/lugares/"+props.tile.title
                                    }}>
                            <InfoIcon/>
                        </IconButton>
                    }
                /> : ""
            }
        </div>
    );
}

export default BannerGrid;