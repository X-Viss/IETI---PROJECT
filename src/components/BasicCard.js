import React from 'react';
import {Card, CardActionArea, CardContent, CardHeader, Typography} from "@material-ui/core";
import FavoriteIcon from '@material-ui/icons/Favorite';

function BasicCard(props) {
    return (
        <Card >
            <CardActionArea>
                <CardHeader
                    action={
                        <FavoriteIcon />
                    }
                    title={props.namesite}
                    subheader={props.location}
                />
                <CardContent>
                    <Typography variant="body2" color="textSecondary" component="p">
                        {props.description}
                    </Typography>
                </CardContent>
            </CardActionArea>
        </Card>
    );
}

export default BasicCard;