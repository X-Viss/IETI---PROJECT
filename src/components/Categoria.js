import React  from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import NewItem from './NewItem.js'
import ItemList from './ItemList.js'
import { Grid } from '@material-ui/core';
import { get, put } from '../requests/axiosRequests.js';
import OptionsBoard from '../common/OptionsBoard';


const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
    width: '100%',
    padding: theme.spacing(0),
    backgroundColor: '#F4F4F4',
    text_transform: "uppercase"
  },
  upperGrid:{
    height: "15vh",
    backgroundColor: "#222A4F",
  }
}));

export default function BodyTabs() {
  const classes = useStyles();
  const queryString = window.location.search;
  const urlParams = new URLSearchParams(queryString);
  const travelId = urlParams.get('travelId');

  const [value, setValue] = React.useState(0);
  const [categories, setCategories] = React.useState([
    {label:'accesorios',
      items:[]},
    {label:'ROPA',
      items:[]},
    {label:'ASEO',
      items:[]},
    {label:'SALUD',
      items:[]},
    {label:'A LA MANO',
      items:[]}
  ]);

  /* istanbul ignore next */
  const loadTravel = () => { /* Cargar Categorias */
    get("bag/travel?travelId="+travelId).then(
        data => {
            setCategories([
              {label:'accesorios',
                items:(data.accessoriesList?data.accessoriesList:[])},
              {label:'ROPA',
                items:(data.clothesList?data.clothesList:[])},
              {label:'ASEO',
                items:(data.cleanlinessList?data.cleanlinessList:[])},
              {label:'SALUD',
                items:(data.medicineList?data.medicineList:[])},
              {label:'A LA MANO',
                items:(data.onHandList?data.onHandList:[])}
            ])
        }
    ).catch((err) => {
        console.log(err)
    })
    console.log("LOADED")
}
  /* istanbul ignore next */
  const saveCategory = (tab) => {
    let list = categories[tab].items;
    switch (tab){
      case 0:put('api/create/category/accessories?id=' + travelId, list); break
      case 1:put('api/create/category/clothes?id=' + travelId, list); break
      case 2:put('api/create/category/cleanliness?id=' + travelId, list); break
      case 3:put('api/create/category/medicine?id=' + travelId, list); break
      case 4:put('api/create/category/onhand?id=' + travelId, list); break
      case 5:put('bag/category/newCategory?travelId=' + travelId, list); break
      default:console.log("WARN")
    }
}

  const handleChange = (event, newValue) => {
    setValue(newValue);
    saveCategory(value); /*Guarda la ultima seleccionada*/ 
    loadTravel();
    console.log("FIN")
  };

  const updateCategory = (values) => { /* Actualizar Item */
    const tab = categories[value].label;
    const itemsTemp = [...categories];
    itemsTemp.forEach(function(cat){
      if(cat.label == tab){
        cat.items.forEach(function(it){
          if(it.name == values.name){
            it.check=!values.check;
          }return it;
        });
      }return cat;
    });
    setCategories(itemsTemp);
    saveCategory(value);
  }

  const extractForm = (values) => { /* Agregar Item */
    const tab = categories[value].label;
    const itemsTemp = [...categories];
    itemsTemp.forEach(function(dato){
      if(dato.label == tab){
        dato.items.push({name:values.value,check:false});
      }return dato;
    });
    setCategories(itemsTemp);
  }

  const handleAddTab = () => {/* Agregar Categoria*/ 
    const catTemp = [...categories];
    catTemp.push({label:'varios',items:[]});
    setCategories(catTemp);
    saveCategory(5);
  }
  console.log("ACTUAL",value)
  return (
    <div className={classes.root}>
      <Grid item xs alignItems="center" className={classes.upperGrid}>
          <OptionsBoard></OptionsBoard>
      </Grid>
      <AppBar position="static" color="default">
        <Tabs
          value={value}
          onChange={handleChange}
          indicatorColor="primary"
          textColor='primary'
          variant="scrollable"
          scrollButtons="auto"
        >
          {categories.map((item, index) => /* header categorias*/
            (
              <Tab key={index} label={item.label} {...scrollBar(index)}/>
            )
          )}
        </Tabs>
      </AppBar>
      {categories.map((item, index) => {
        return (
        <TabPanel key={index} value={value} index={index}>
          <ItemList currentTab={value} onCheckBox={updateCategory} travel={travelId} items={item.items}></ItemList>
        </TabPanel>
        )
      })}
      <NewItem value={value} onSubmitForm={extractForm} onSubmitTab={handleAddTab} ></NewItem>
    </div>
  );
}

function scrollBar(index) {
  return {
    id: `scrollable-auto-tab-${index}`,
    'aria-controls': `scrollable-auto-tabpanel-${index}`,
  };
}

function TabPanel(props) {
  const { children, value, index, ...other } = props;

  return (
    <Typography
      component="div"
      role="tabpanel"
      hidden={value !== index}
      id={`scrollable-auto-tabpanel-${index}`}
      aria-labelledby={`scrollable-auto-tab-${index}`}
      {...other}
    >
      {value === index && <Box>{children}</Box>}
    </Typography>
  );
}