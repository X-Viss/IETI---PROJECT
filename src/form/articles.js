import React from 'react'

export function ArticleForForm(props){
    const {item,className}=props
    const classNameToSet = className? className:"noneClass"
    return <article className={classNameToSet}>
        {item}
    </article>
}

export function MapArticles(props){
    const {items} = props
    return items.map((item,index)=>{
        return <ArticleForForm item={item.item} className={item.className} key={index}/>
    })
}