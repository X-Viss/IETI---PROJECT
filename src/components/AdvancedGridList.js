import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import GridList from '@material-ui/core/GridList';
import GridListTile from '@material-ui/core/GridListTile';
import {Typography} from "@material-ui/core";
import BannerGrid from "./BannerGrid";

const useStyles = makeStyles((theme) => ({
    root: {
        display: 'flex',
        flexWrap: 'wrap',
        justifyContent: 'space-around',
        overflow: 'hidden',
        backgroundColor: theme.palette.background.paper,
    },
    gridList: {
        //width: "100%",
        //height: 450,
        // Promote the list into his own layer on Chrome. This cost memory but helps keeping high FPS.
        transform: 'translateZ(0)',
    },
    titleBar: {
        background:
            'linear-gradient(to bottom, rgba(0,0,0,0.7) 0%, ' +
            'rgba(0,0,0,0.3) 70%, rgba(0,0,0,0) 100%)',
    },
    icon: {
        color: 'rgba(255, 255, 255, 0.54)',
    },
}));

export default function AdvancedGridList(props) {
    const classes = useStyles();

    return (
        <div className={classes.root} style={{backgroundColor: "#F4F4F4"}}>
            <Typography variant="h3" gutterBottom style={{color: '#222A4F'}} onClick={event => {
                event.preventDefault();
                window.location.href = props.href
            }}>
                {props.title}
            </Typography>
            <GridList cellHeight={200} spacing={1} cols={props.numcols ? props.numcols : 2}
                      className={classes.gridList}>
                {props.tileData.map((tile) => (
                    <GridListTile key={tile.img} cols={tile.featured}>
                        <BannerGrid tile={tile} classes={classes}/>
                    </GridListTile>
                ))}
            </GridList>
        </div>
    );
}
