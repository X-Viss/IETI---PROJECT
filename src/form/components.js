import React from 'react'
import { MapArticles } from './articles'
import TextField from '@material-ui/core/TextField'
import Autocomplete from '@material-ui/lab/Autocomplete'
import './login.css';

export function itemsForRegisterForm() {
    const confirmPasswords = (event) => {
        const password = document.getElementsByName('password')[0]
        const password2 = document.getElementsByName('password2')[0]
        if (password.value === password2.value) {
            password.setCustomValidity("")
            password2.setCustomValidity("")
        } else {
            password.setCustomValidity("Las Contraseñas deben ser iguales")
            password2.setCustomValidity("Las Contraseñas deben ser iguales")
        }

    }
    const countries = [
        { code: 'AD', label: 'Andorra', phone: '376' },
    ]
    return [
        {
            className: "",
            item: <TextField label="Correo" type="email" name="correo" required={true} />
        },
        {
            className: "",
            item: <TextField label="Contraseña" type="password" name="password" required={true} onChange={confirmPasswords} />
        },
        {
            className: "",
            item: <TextField label="Confirmar Contraseña" type="password" name="password2" required={true} onChange={confirmPasswords} />
        },
        {
            className: "",
            item: <Autocomplete
                id="country"
                name="country"
                options={countries}
                getOptionLabel={(option) => option.label}
                style={{ width: 300 }}

                renderInput={(params) => <TextField {...params} required={true} label="Pais" variant="outlined" />}
            />
        },
        {
            className: "",
            item: <TextField
                id="date"
                label="Birthday"
                type="date"
                name="date"
                required={true}
                InputLabelProps={{
                    shrink: true,
                }}
            />
        },
        {
            className: "",
            item: <TextField label="Telefono" type="number" name="phone" required={true} />
        },
    ]
}

export function RegisterForm(props) {
    const handleSubmit = (event) => {
        event.preventDefault()
        console.log(event.target.elements)
    }
    const items = itemsForRegisterForm()
    return <div className="container">
        <div className="sing-in">
            <div className="sign-in-container">
                <form onSubmit={handleSubmit}>
                    <center>
                        <h1>Registro</h1>
                        <p/>
                        <MapArticles items={items} />
                        <p/>
                        <button type="submit" variant="contained">Submit</button>
                        <p/><p/>
                    </center>
                    <button type="submit" className="ghost">Ya tienes cuenta? Ingresa</button>
                    </form>
            </div>
        </div>
        <div className="overlay-container">
            <div className="overlay overlay-right">
            </div>
	    </div>
    </div>
}

export function itemsForLoginForm(){
    return[
        {
            className:"",
            item:<TextField label="Usuario" type="email" name="usuario" required={true}/>
        },
        {
            className:"",
            item:<TextField label="Contraseña" type="password" name="password" required={true}/>
        }
    ]
}


export function LoginForm(props){
    const handleSubmit= (event)=>{
        event.preventDefault()
        console.log('submit')
    }
    return <div className="container">
        <div className= "sing-in">
            <div className="sign-in-container">
                <form onSubmit={handleSubmit}>
                    <center>
                    <h1>Sign in</h1>
                    <p/>
                    <MapArticles items={itemsForLoginForm()}/>
                    <p/>
                    <button type="submit">Login</button>
                    <p/><p/>
                    </center>
                    <button type="submit" className="ghost">No tienes cuenta? Registrate</button>
                </form>
            </div>
        </div>
        <div className="overlay-container">
            <div className="overlay overlay-right">
            </div>
	    </div>
    </div>
}