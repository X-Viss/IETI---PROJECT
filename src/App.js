import './App.css';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { RegisterForm } from './form/components';
import { Travels } from './travel';
import { LoginForm } from './form/components';
import { NotFoundForm } from './form/404';
import Main from './seleccionarClima/decoracionMain'
import { EditUserForm } from './form/edit';
import { useEffect } from 'react';
import { useState } from 'react';
import ProtectedRoute from './tests/common/PrivateRoute';
import Inicio from "./inicio/Inicio";
import Lugares from "./lugares/lugares";
import Tiposviajes from "./tiposviajes/tiposviajes";
import Statisctics from "./statisctics/statisctics";

function App() {

  const [loggedIn, setLoggedIn] = useState(localStorage.getItem("loggedIn")=="true");
  
  useEffect(() => {
    const logged = localStorage.getItem("loggedIn");
    if (logged) {
      console.log("DONE")
      setLoggedIn(logged == "true");
      console.log(loggedIn)
    }
  });

  return (
    <div className="App">

      <Router>
        <Switch>
          <Route exact path="/inicio"><Inicio/></Route>
          <Route exact path="/lugares"><Lugares/></Route>
          <Route exact path="/tiposviajes"><Tiposviajes/></Route>
          <Route exact path="/statisctics"><Statisctics/></Route>
          <Route exact path="/registro" ><RegisterForm /></Route>
          <Route exact path="/login" ><LoginForm /></Route>
          <ProtectedRoute exact isAuthenticated={loggedIn} path="/travelList" component={() => (<Travels/>)} />
          <ProtectedRoute exact isAuthenticated={loggedIn} path="/categorias" component={() => (<Main/>)} />
          <ProtectedRoute exact isAuthenticated={loggedIn} path="/editUser" component={() => (<EditUserForm />)} />
          <Route><NotFoundForm /></Route>

        </Switch>
      </Router>


    </div>
  );
}

export default App;