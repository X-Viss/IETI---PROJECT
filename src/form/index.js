import {LoginForm} from './components'

export{
    LoginForm
}