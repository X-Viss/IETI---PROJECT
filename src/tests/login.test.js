import Enzyme, { mount, shallow } from 'enzyme'
import { createRender } from '@material-ui/core/test-utils';
import { itemsForRegisterForm, LoginForm } from '../form/components';
import { ArticleForForm, MapArticles } from '../form/articles';
import App from '../App';


it('The app renders', () => {
  shallow(<App />)
})

it('The register renders', () => {
  shallow(<LoginForm />)
})

it('The login renders map', () => {
  const prueba = [
    {
      className: "",
      item: <p label="Usuario" type="email" name="usuario" required={true} />
    },
    {
      className: "",
      item: <p label="ContraseÃ±a" type="password" name="password" required={true} />
    }
  ]
  shallow(<MapArticles items={prueba} />)
})

it('The register renders items', () => {
  const arr = itemsForRegisterForm()
  expect(arr).toHaveLength(6)
})

it('The login renders article', () => {
  const prueba = [
    {
      className: "",
      item: <p label="Usuario" type="email" name="usuario" required={true} />
    }
  ]
  shallow(<ArticleForForm item={prueba.item} className={prueba.className} />)
})

it('The login renders article not class', () => {
  const prueba = [
    {
      item: <p label="Usuario" type="email" name="usuario" required={true} />
    }
  ]
  const wrapper = mount(<ArticleForForm item={prueba.item} className={"hola"} />)
  expect(wrapper.find('article.hola').instance().className).toBe('hola')

})

it('calls click event not do noting', () => {
  const wrapper = mount(<LoginForm />);
  wrapper.find('button').at(0).simulate('click');
  const vari = wrapper.find('button').at(0).simulate('click');
  expect(vari).toEqual({});
});

describe('<LoginForm />', () => {
  let render;

  beforeAll(() => {
    render = createRender();
  });

  it('should work', () => {
    render(<LoginForm />);
  });
});

it('calls click event', () => {
  const wrapper = mount(<LoginForm />);
  wrapper.children().find('input[type="password"]').simulate('change', {target: {name: 'password', value: '123'}});
  wrapper.find('input[name="usuario"]').simulate('change',{currentTarget:{value:'algo@algo.com'}})
  wrapper.find('form').simulate('submit')
  expect(wrapper.find('input[name="usuario"]').instance().value).toEqual('');
});
