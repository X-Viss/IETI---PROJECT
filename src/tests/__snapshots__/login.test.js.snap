// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The app renders 1`] = `
<div
  className="App"
/>
`;

exports[`The login renders 1`] = `
<div
  className="container"
>
  <div
    className="sing-in"
  >
    <div
      className="sign-in-container"
    >
      <form
        onSubmit={[Function]}
      >
        <center>
          <h1>
            Sign in
          </h1>
          <p />
          <MapArticles
            items={
              Array [
                Object {
                  "className": "",
                  "item": <WithStyles(ForwardRef(TextField))
                    label="Usuario"
                    name="usuario"
                    required={true}
                    type="email"
                  />,
                },
                Object {
                  "className": "",
                  "item": <WithStyles(ForwardRef(TextField))
                    label="Contraseña"
                    name="password"
                    required={true}
                    type="password"
                  />,
                },
              ]
            }
          />
          <p />
          <button
            type="submit"
          >
            Login
          </button>
          <p />
          <p />
        </center>
        <button
          className="ghost"
          type="submit"
        >
          No tienes cuenta? Registrate
        </button>
      </form>
    </div>
  </div>
  <div
    className="overlay-container"
  >
    <div
      className="overlay overlay-right"
    />
  </div>
</div>
`;

exports[`The login renders article 1`] = `
<article
  className="noneClass"
/>
`;

exports[`The login renders map 1`] = `
Array [
  <ArticleForForm
    className=""
    item={
      <p
        label="Usuario"
        name="usuario"
        required={true}
        type="email"
      />
    }
    key="0"
  />,
  <ArticleForForm
    className=""
    item={
      <p
        label="Contraseña"
        name="password"
        required={true}
        type="password"
      />
    }
    key="1"
  />,
]
`;
